### PR TITLE
Bug 1816908 - The search via the widget's microphone should open a w tab containing the search term.

### PR DESCRIPTION
**Pull request by iorgamgabriel**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1816908

### CODE REVIEW

**Comment by iorgamgabriel** [comment]
https://user-images.githubusercontent.com/8118371/219060423-cbc6302d-8d30-4463-b3fc-35f843952bde.mp4

File: focus-android/app/src/main/java/org/mozilla/focus/activity/MainActivity.kt [hunk 1]
```diff
@@ -281,21 +282,25 @@ open class MainActivity : LocaleAwareAppCompatActivity() {
...
-            is Screen.SitePermissionOptionsScreen -> components.appStore.dispatch(
-                AppAction.OpenSitePermissionOptionsScreen(
-                    sitePermission =
-                    (components.appStore.state.screen as Screen.SitePermissionOptionsScreen).sitePermission,
-                ),
-            )
-            else -> {
-                handleAppNavigation(intent)
+        if (!intent.extras?.getString(BaseVoiceSearchActivity.SPEECH_PROCESSING).isNullOrEmpty()) {
+            handleAppNavigation(intent)
```
Comment (Mugurell) [inline focus-android/app/src/main/java/org/mozilla/focus/activity/MainActivity.kt:286]:
```suggestion
        if (!intent.extras?.getString(BaseVoiceSearchActivity.SPEECH_PROCESSING).isNullOrEmpty()) {
            handleAppNavigation(intent)
            return
        }
```

**Review by Mugurell — APPROVED** [review]
LGTM. Thanks for the quick fix!
Left a suggestion for maybe simplifying the if-else branching.

File: focus-android/app/src/main/java/org/mozilla/focus/activity/MainActivity.kt (hunk 1 above)
Comment (iorgamgabriel) [reply focus-android/app/src/main/java/org/mozilla/focus/activity/MainActivity.kt:286]: Done

**Comment by iorgamgabriel** [comment]
@Mergifyio backport releases_v111.0

**Comment by iorgamgabriel** [comment]
@Mergifyio backport releases_v111

**Pull request merged**